client/ayon_shotgrid/addon.py

```diff
@@ -80,6 +80,18 @@ def is_local_storage_enabled(self):
     def get_local_storage_key(self):
         return self._local_storage_key or None
 
+    def get_credentials(self):
+        from .lib import credentials
+
+        if self._client_login_type == "env":
+            user_login = os.getenv("AYON_SG_USERNAME")
```

Comment:
```suggestion 
            user_login = os.getenv("AYON_SG_USERNAME") or os.getenv("USER")
```

Reply:
ehm.. like this ? 

```suggestion 
            user_login = os.getenv("AYON_SG_USERNAME") or os.getenv("AYON_USERNAME")
```

Reply:
no, I'm saying that we should check which env var is the one is being set across the different addons for logging in. I believe it might be we are just setting `AYON_USERNAME` now instead of `AYON_SG_USERNAME`. It would make more sense it's `AYON_USERNAME`

Reply:
@iLLiCiTiT could you tell me which env var is recommended to get the current user name ?

Reply:
@MustafaJafar it should be pretty easy to check, just check the env vars that are present in your environment from any DCC and also check the env vars that are passed to Deadline when you submit a job

Reply:
```python
from ayon_core.lib import get_ayon_username
```

```suggestion 
            user_login = get_ayon_username()
```

Reply:
you haven't applied this change yet have you?

Reply:
As far as I know, this change will get the ayon user name which is not the same as sg name (login mail).
Also, we gave this PR a test run and faced an issue https://github.com/ynput/ayon-shotgrid/pull/118#issuecomment-2382987181

Reply:
oh, for us the sg name is the same as the ayon user name, we don't use the mail as `login` in SG, that's why there's a separate `email` field:
<img width="619" alt="Screenshot 2024-10-02 at 9 28 12 AM" src="https://github.com/user-attachments/assets/ed50d56f-fa8f-47da-823b-4730d23aa7ae">

Reply:
I think my ayon user name is not the same as SG name. I don't have them synced and also because I'm using an AYON instance for dev, I mostly use the default `Admin` username 
![image](https://github.com/user-attachments/assets/32221019-5da5-459f-93c3-8ff9a748a3bf)

Reply:
yeah it doesn't look like, but you could change that `login` field on SG to match if you wanted

Reply:
I feel like for people who's SG set up doesn't have that on sync it doesn't make sense to use this login type so it should be documented the different types of logins and when to use one or another based on your set up

Reply:
marking this as resolved as mentioned in https://github.com/ynput/ayon-shotgrid/pull/118#pullrequestreview-2415867311